src/main/java/at/hannibal2/skyhanni/features/chat/PowderMiningChatFilter.kt

```diff
@@ -0,0 +1,376 @@
+            pickonimbusPattern to PICKONIMBUS to "powder_mining_pickonimbus",
+            jungleHeartPattern to JUNGLE_HEART to "powder_mining_jungle_heart",
+            sludgeJuicePattern to SLUDGE_JUICE to "powder_mining_sludge_juice",
+            yoggiePattern to YOGGIE to "powder_mining_yoggie",
+            robotPartsPattern to ROBOT_PARTS to "powder_mining_robot_parts",
+            treasuritePattern to TREASURITE to "powder_mining_treasurite",
+        )
+        for ((patternToReward, returnReason) in rewardPatterns) {
+            if (patternToReward.first.matches(ssMessage)) {
+                return if (config.simplePowderMiningTypes.contains(patternToReward.second)) returnReason
+                else "no_filter"
+            }
+        }
+
+        //Goblin Eggs
+        goblinEggPattern.matchMatcher(ssMessage) {
+            if (config.goblinEggs != PowderMiningFilterConfig.GoblinEggFilterEntry.SHOW_ALL) {
+                if (config.goblinEggs == PowderMiningFilterConfig.GoblinEggFilterEntry.HIDE_ALL) return "powder_mining_goblin_eggs"
+
+                val colorStr = groupOrNull("color")?.lowercase() ?: ""
+                when (colorStr) {
+                    //'Colorless', base goblin eggs will never be shown in this code path
+                    "" -> return "powder_mining_goblin_eggs"
+                    "green" -> return if (config.goblinEggs > PowderMiningFilterConfig.GoblinEggFilterEntry.GREEN_UP) {
+                        return "powder_mining_goblin_eggs"
+                    } else "no_filter"
+                    "yellow" -> return if (config.goblinEggs > PowderMiningFilterConfig.GoblinEggFilterEntry.YELLOW_UP) {
+                        "powder_mining_goblin_eggs"
+                    } else "no_filter"
+                    "red" -> return if (config.goblinEggs > PowderMiningFilterConfig.GoblinEggFilterEntry.RED_UP) {
+                        "powder_mining_goblin_eggs"
+                    } else "no_filter"
+                    // BLUE_ONLY enum not explicitly used in comparison, as the only
+                    // case that will block it is HIDE_ALL, which is covered above
+                    "blue" -> return "no_filter"
+                    else -> {
+                        ChatUtils.chat(
+                            "§cUnknown Goblin Egg color detected in Powder Mining Filter: " +
+                                "'${colorStr}' - please report this in the Discord!",
+                        )
+                        return "no_filter"
+                    }
+                }
+            } else return "no_filter"
+        }
+
+        //Gemstones
+        gemstonePattern.matchMatcher(ssMessage) {
+            val gemStr = groupOrNull("gem")?.lowercase() ?: ""
+            val tierStr = groupOrNull("tier")?.lowercase() ?: ""
+
+            //Theoretically impossible but ?
+            if (gemStr.isEmpty() || tierStr.isEmpty()) return ""
+
+            val gemSpecificConfig = when (gemStr) {
```

Comment:
use a different name. this is not the config, but the setting. the suffix "config" in skyhanni means the actual config element, not the value in it